page-archives.php

```diff
@@ -362,20 +362,30 @@ function insertSponsors($bcbname)
                     <div id="archives_page_title"><?php the_title();?></div>
                     <div id="archives_page_bcb_selector_buttons">
                         <ul id="archives_page_bcb_selector">
-                            <li class="archive_selected"><a class="archives_current" data-contentdiv="archive_1691">BCB Spring 2018</a></li>
-                            <li><a data-contentdiv="archive_1556">BCB Spring 2016</a></li>
-                            <li><a data-contentdiv="archive_1458">BCB Monsoon 2015</a></li>
-                            <li><a data-contentdiv="archive_1366">BCBX AntHill</a></li>
-                            <li><a data-contentdiv="archive_1224">BCB17</a></li>
-                            <li><a data-contentdiv="archive_1057">BCB16</a></li>
-                            <li><a data-contentdiv="archive_931">BCB15</a></li>
-                            <li><a data-contentdiv="archive_785">BCB14</a></li>
-                            <li><a data-contentdiv="archive_636">BCB13</a></li>
-                            <li><a data-contentdiv="archive_479">BCB12</a></li>
-                            <li><a data-contentdiv="archive_399">BCB11</a></li>
-                            <li><a data-contentdiv="archive_324">BCB10</a></li>
-                            <li><a data-contentdiv="archive_220">BCB9</a></li>
-                            <li><a data-contentdiv="archive_3">BCB8</a></li>
+                            <?php
+                            $first = true;
+                            $archive_term_id = get_cat_ID('archives');
+                            $taxonomy_name = 'category';
+                            $termchildren = get_term_children( $archive_term_id, $taxonomy_name );
+                            echo '<ul>';
+                            foreach ( $termchildren as $child ) {
+                                $term = get_term_by( 'id', $child, $taxonomy_name );
+                                // Get the depth of the category.
+                                $cats_str = get_category_parents($term, false, '%#%');
+                                $cats_array = explode('%#%', $cats_str);
+                                $cat_depth = sizeof($cats_array)-2;
```

Comment:
This -2 is unnecessary and misleading as its changing the depth value for rest of the code setting wrong expectations for a future dev using it later in code. Specially if the variable is named cat_depth it should carry the correct depth value of the cat. I dont see any specific reason for using the value 1. You might as well check value 3 in the following if condition, or change the name of the variable to something more contextual.